hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAbfsInputStreamStatistics.java

```diff
@@ -399,6 +406,89 @@ public void testActionHttpGetRequest() throws IOException {
     }
   }
 
+  @Test
+  public void testPread() throws IOException {
+    describe("Testing preads in AbfsInputStream");
+
+    AzureBlobFileSystem fs = getFileSystem();
+    AzureBlobFileSystemStore abfss = fs.getAbfsStore();
+    Path readStatPath = path(getMethodName());
+
+    AbfsOutputStream out = null;
+    AbfsInputStream in = null;
+    try {
+      out = createAbfsOutputStreamWithFlushEnabled(fs, readStatPath);
+      /*
+       * Writing 1MB buffer to the file.
+       */
+      out.write(defBuffer);
+      out.hflush();
+
+      in = abfss.openFileForRead(readStatPath, Optional.empty(),
```

Comment:
if you open this before L424, you'd also verify that openFile itself isn't doing a buffered read on open, wouldn't you?

Reply:
Moved stat asserts to new test file as per another comment.